File: algorithms/code/leetcode/lc027_remove_element/LC027RemoveElement.Tests.cs
using System.Linq;
using Xunit;

namespace AlgorithmDesign.code.leetcode.lc027_remove_element
{
    public class SolutionTests
    {
        [Fact]
        public void test_0_42_1_2_42_3_4__42()
        {
            int[] nums = { 0, 42, 1, 2, 42, 3, 4 };
            int length = new Solution().RemoveElement(nums, 42);
            Assert.Equal(5, length);
            int[] expected = { 0, 1, 2, 3, 4 };
            Assert.Equal(expected, nums.Take(length));
        }

        [Fact]
        public void test_nothing()
        {
            int length = new Solution().RemoveElement(null, 42);
            Assert.Equal(0, length);
        }
    }
}

File: algorithms/code/leetcode/lc027_remove_element/LC027RemoveElementTests.java
package leetcode.lc027_remove_element;

import java.util.Arrays;

import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.assertArrayEquals;
import static org.junit.jupiter.api.Assertions.assertEquals;

public final class LC027RemoveElementTests {
    @Test
    public void test042124234And42() throws Exception {
        LC027RemoveElement solution;
        solution = new LC027RemoveElement();
        int[] nums = { 0, 42, 1, 2, 42, 3, 4 };
        int length = solution.removeElement(nums, 42);
        assertEquals(5, length);
        int[] expected = { 0, 1, 2, 3, 4 };
        assertArrayEquals(expected, Arrays.copyOfRange(nums, 0, length));
    }

    @Test
    public void testNothing() throws Exception {
        LC027RemoveElement solution;
        solution = new LC027RemoveElement();
        int length = solution.removeElement(null, 42);
        assertEquals(0, length);
    }
}
